tests/judges/test_instruction_judge.py

```diff
@@ -0,0 +1,298 @@
+        mock_judge = Mock()
+        mock_registered = Mock()
+        mock_judge.register.return_value = mock_registered
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(name='No Exp Judge', user_instructions='Test without experiment')
+
+        result = judge.register_scorer()
+
+        # Verify register was called without experiment_id
+        mock_judge.register.assert_called_once()
+        call_args = mock_judge.register.call_args
+        self.assertIn('name', call_args.kwargs)
+        self.assertNotIn('experiment_id', call_args.kwargs)
+
+    @patch('server.judges.instruction_judge.make_judge')
+    @patch('server.judges.instruction_judge.logger')
+    def test_register_scorer_error_handling(self, mock_logger, mock_make_judge):
+        """Test register_scorer handles errors gracefully."""
+        mock_judge = Mock()
+        mock_judge.register.side_effect = Exception("Registration failed")
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(
+            name='Error Register Judge', user_instructions='This registration will fail'
+        )
+
+        result = judge.register_scorer()
+
+        # Should return None and log warning
+        self.assertIsNone(result)
+        mock_logger.warning.assert_called_once()
+
+    @patch('server.judges.instruction_judge.make_judge')
+    def test_optimize_with_sufficient_data(self, mock_make_judge):
+        """Test optimize method with sufficient traces."""
+        mock_judge = Mock()
+        mock_aligned_judge = Mock()
+        mock_judge.align.return_value = mock_aligned_judge
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(name='Optimize Test Judge', user_instructions='Test optimization')
+
+        # Create mock traces (need at least 10 for MIN_EXAMPLES_FOR_OPTIMIZATION)
+        traces = [Mock() for _ in range(12)]
+
+        result = judge.optimize(traces)
+
+        # Verify alignment was called and judge was updated
+        mock_judge.align.assert_called_once_with(traces=traces)
+        self.assertEqual(judge.scorer_func, mock_aligned_judge)
+        self.assertTrue(result)
+
+    @patch('server.judges.instruction_judge.make_judge')
+    @patch('server.judges.instruction_judge.logger')
+    def test_register_scorer_failure_returns_none(self, mock_logger, mock_make_judge):
+        """Test register_scorer returns None on failure."""
+        mock_judge = Mock()
+        mock_judge.register.side_effect = Exception("Registration failed")
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(
+            name='Registration Fail Judge', user_instructions='Test registration failure'
+        )
+
+        result = judge.register_scorer()
+
+        # Should return None and log warning
+        self.assertIsNone(result)
+        mock_logger.warning.assert_called_once()
```

Comment:
This tests an implementation detail IMO. If we want to test that a specific warning is emitted then we should include the text in the assertion.